doc/source/tutorial/stats/sampling_pinv.rst

```diff
@@ -0,0 +1,271 @@
+
+.. math::
+    \epsilon_{u}(u) = | u - F\left(F^{-1}_{a}(u)\right) |
+
+where :math:`u \in (0, 1)` is a quantile where we want to measure the error
+and :math:`F^{-1}_a` is the approximated PPF of the given distribution.
+
+The maximal u-error is the criterion for approximation errors when calculating
+the CDF and PPF numerically. The maximal tolerated u-error of an algorithm is
+called the u-resolution of the algorithm and denoted by :math:`\epsilon_{u}`:
+
+.. math::
+    \sup_{u \in (0,1)} | u - F\left(F^{-1}_{a}(u)\right) | \le \epsilon_{u}
+
+There are two drawbacks of using u-resolution:
+
+* Does not work for distributions with high and narrow peaks
+* the CDF approximation error is high in the tails of the distribution
+
+Also, the method only works for bounded distributions. In case of infinite
+tails, the ends of the tails are cut off such that the area under them is
+less than or equal to :math:`0.05\epsilon_{u}`.
+
+Following four steps are carried out by the algorithm during setup:
+
+* Computing the end points of the distribution: If a finite support is given,
+  this step is skipped. Otherwise, the ends of the tails are cut off such that
+  the area under them is less than or equal to :math:`0.05\epsilon_{u}`.
+* The domain is divided into subintervals to compute the CDF and PPF.
+* The CDF is computed using Gauss-Lobatto quadrature such that the integration
+  error is at most :math:`0.05I_{0}\epsilon_{u}` where :math:`I_{0}` is
+  approximately the total area under the PDF.
+* The PPF is computed using Newton's interpolating formula with maximum
+  interpolation error :math:`0.9\epsilon_{u}`.
+
+To initialize the generator to sample from a stnadard normal distribution, do:
+
+    >>> from scipy.stats import NumericalInversePolynomial
+    >>> class StandardNormal:
+    ...     def pdf(self, x):
+    ...         return np.exp(-0.5 * x*x)
+    ...
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInversePolynomial(dist, random_state=urng)
+
+The generator has been setup and we can start sampling from our distribution:
+
+    >>> rng.rvs((5, 3))
+    array([[-1.52449963,  1.31933688,  2.05884468],
+           [ 0.48883931,  0.15207903, -0.02150773],
+           [ 1.11486463,  1.95449597, -0.30724928],
+           [ 0.9854643 ,  0.29867424,  0.7560304 ],
+           [-0.61776203,  0.16033378, -1.00933003]])
+
+We can look at the histogram of the random variates to check how well they fit
+our distribution:
+
+.. plot::
+
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.stats import norm
+    >>> from scipy.stats import NumericalInversePolynomial
+    >>> class StandardNormal:
+    ...     def pdf(self, x):
+    ...         return np.exp(-0.5 * x*x)
+    ...
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInversePolynomial(dist, random_state=urng)
+    >>> rvs = rng.rvs(100000)
```

Comment:
maybe that number can be reduced quite a bit since we only use 50 bins, e.g. I guess the histogram still looks ok with ize=10,000?

Reply:
> I guess the histogram still looks ok with ize=10,000?

Yes, it does. Will change it this way.

Reply:
Done.